streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -189,14 +185,15 @@
         );
     }
 
-    private <KR, T> KTable<KR, T> doAggregate(final KStreamAggProcessorSupplier<K, KR, V, T> aggregateSupplier,
-                                              final String functionName,
-                                              final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materializedInternal) {
+    private <T> KTable<K, T> doAggregate(final KStreamAggProcessorSupplier<K, K, V, T> aggregateSupplier,
```

Comment:
The reason of this is because `KStreamAggProcessorSupplier` is used for both un-windowed aggregates (here) and windowed aggregates. For the latter, `<RK>` would be `Windowed<K>`.

Reply:
Ack.